x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -332,6 +352,115 @@ private void createApiKeyAndIndexIt(
         }));
     }
 
+    public void updateApiKey(
+        final Authentication authentication,
+        final UpdateApiKeyRequest request,
+        final Set<RoleDescriptor> userRoles,
```

Comment:
Btw, I am _not_ suggestion having the refactoring as part of the "Updatable API keys" work stream. By "get back to it in future", I meant some point after we deliver the planned feature. So no need to bundle it in the PR for REST layer as it might add unnecessary complexities.